certification/pyxis/types.go

```diff
@@ -1,10 +1,201 @@
 package pyxis
 
 import (
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"path"
+
+	"github.com/redhat-openshift-ecosystem/openshift-preflight/certification/errors"
 	"github.com/redhat-openshift-ecosystem/openshift-preflight/certification/formatters"
 )
 
-type CertificationInput struct {
+// certificationInputBuilder facilitates the building of CertificationInput for
+// submitting an asset to Pyxis.
+type certificationInputBuilder struct {
+	certificationInput
+}
+
+// NewCertificationInput accepts required values for submitting to Pyxis, and returns a CertificationInputBuilder for
+// adding additional files as artifacts to the submission.
+func NewCertificationInput(project *CertProject, rpmManifestFilePath, certImageFilepath, preflightResultsFilepath string) (*certificationInputBuilder, error) {
```

Comment:
We discussed this at length. I maintain that methods should be near the types (for reading purposes). Godoc, for example, makes it so that's not necessary from a code exploration point of view, so I'm flexible. I'm going to leave it as is for now, but we may want to revisit how we define types in general. We might be able to see some benefit of using types packages in certain strategic locations.

Reply:
If types.go gets huge in the future, we can revisit. For now, it's minimal impact.

Reply:
@bcrochet I went ahead and moved the Builder initializer and its type to builder.go, since it doesn't really serve the same purpose as all the other types in types.go.